Sources/_StringProcessing/ConsumerInterface.swift

```diff
@@ -0,0 +1,602 @@
+    case .emojiModifier:
+      break // availability
+    case .emoji:
+      break // availability
+    case .emojiPresentation:
+      break // availability
+    case .extender:
+      return consumeScalarProp(\.isExtender)
+    case .extendedPictographic:
+      break
+    case .fullCompositionExclusion:
+      return consumeScalarProp(\.isFullCompositionExclusion)
+    case .graphemeBase:
+      return consumeScalarProp(\.isGraphemeBase)
+    case .graphemeExtended:
+      break
+    case .graphemeLink:
+      break
+    case .hexDigit:
+      return consumeScalarProp(\.isHexDigit)
+    case .hyphen:
+      break
+    case .idContinue:
+      return consumeScalarProp(\.isIDContinue)
+    case .ideographic:
+      return consumeScalarProp(\.isIdeographic)
+    case .idStart:
+      return consumeScalarProp(\.isIDStart)
+    case .idsBinaryOperator:
+      return consumeScalarProp(\.isIDSBinaryOperator)
+    case .idsTrinaryOperator:
+      return consumeScalarProp(\.isIDSTrinaryOperator)
+    case .joinControl:
+      return consumeScalarProp(\.isJoinControl)
+    case .logicalOrderException:
+      return consumeScalarProp(\.isLogicalOrderException)
+    case .lowercase:
+      return consumeScalarProp(\.isLowercase)
+    case .math:
+      return consumeScalarProp(\.isMath)
+    case .noncharacterCodePoint:
+      return consumeScalarProp(\.isNoncharacterCodePoint)
+    case .otherAlphabetic:
+      break
+    case .otherDefaultIgnorableCodePoint:
+      break
+    case .otherGraphemeExtended:
+      break
+    case .otherIDContinue:
+      break
+    case .otherIDStart:
+      break
+    case .otherLowercase:
+      break
+    case .otherMath:
+      break
+    case .otherUppercase:
+      break
+    case .patternSyntax:
+      return consumeScalarProp(\.isPatternSyntax)
+    case .patternWhitespace:
+      return consumeScalarProp(\.isPatternWhitespace)
+    case .prependedConcatenationMark:
+      break
+    case .quotationMark:
+      return consumeScalarProp(\.isQuotationMark)
+    case .radical:
+      return consumeScalarProp(\.isRadical)
+    case .regionalIndicator:
+      break
+    case .softDotted:
+      return consumeScalarProp(\.isSoftDotted)
+    case .sentenceTerminal:
+      return consumeScalarProp(\.isSentenceTerminal)
+    case .terminalPunctuation:
+      return consumeScalarProp(\.isTerminalPunctuation)
+    case .unifiedIdiograph: // spelling?
+      return consumeScalarProp(\.isUnifiedIdeograph)
+    case .uppercase:
+      return consumeScalarProp(\.isUppercase)
+    case .variationSelector:
+      return consumeScalarProp(\.isVariationSelector)
+    case .whitespace:
+      return consumeScalarProp(\.isWhitespace)
+    case .xidContinue:
+      return consumeScalarProp(\.isXIDContinue)
+    case .xidStart:
+      return consumeScalarProp(\.isXIDStart)
+    case .expandsOnNFC:
+      break
+    case .expandsOnNFD:
+      break
+    case .expandsOnNFKC:
+      break
+    case .expandsOnNFKD:
+      break
```

Comment:
These properties are all deprecated as of Unicode 6 (we're on 14 now). Do we need to support these?

Reply:
We might one day.... but for now we'll punt on these